### PR TITLE
fix: added optional separator line to menu list items

### PR DESCRIPTION
**Pull request by kavya-b**
#### Please provide a link to the associated issue.
#1133 
#### Please provide a brief summary of this pull request.
Added optional separator line to menu list items. On fd-menu tag, use **[separator] = "true"** to get the line separator on the list items.
#### If this is a new feature, have you updated the documentation?
Yes.

### CODE REVIEW

File: library/src/lib/menu/menu.component.ts [hunk 1]
```diff
@@ -12,9 +15,24 @@ import {
     encapsulation: ViewEncapsulation.None,
     styleUrls: ['menu.component.scss']
 })
-export class MenuComponent {
+export class MenuComponent extends AbstractFdNgxClass {
```
Comment (JKMarkowski) [inline library/src/lib/menu/menu.component.ts]: Could you please use `@HostBinding` along with `@Input` to bind class to input variable? There are some examples in code. There is no sense to use `AbstractFdNgxClass` anymore.

File: library/src/lib/menu/menu.component.ts [hunk 2]
```diff
@@ -12,9 +15,24 @@ import {
...
+    separator: boolean = false;
+
+    /** @hidden */
+    constructor(public itemEl: ElementRef) {
+        super(itemEl);
+    }
+    /** @hidden */
+    _setProperties(): void {
+        if (this.separator) {
+            this._addClassToElement('fd-menu__list-separator');
```
Comment (JKMarkowski) [inline library/src/lib/menu/menu.component.ts]: Also this class shouldn't be added there. We need to add class `fd-menu__list--separated` into `fd-menu-list` component.

File: library/src/lib/menu/menu.component.scss [hunk 3]
```diff
@@ -3,3 +3,7 @@
         outline: var(--fd-color-action-focus) dotted 1px;
     }
 }
+.fd-menu__list-separator > ul > li,
+.fd-menu__list-separator > fd-menu-group > ul > li{
+    border-bottom: 1px #EDEDED solid;
+}
```
Comment (JKMarkowski) [inline library/src/lib/menu/menu.component.scss]: The General problem of this solve is that, we would need to follow all of the changes from fundamental styles. So let's use prepared class and delete this one.

**Comment by kavya-b** [comment]
Hi @JKMarkowski , as discussed, have made changes in accordance to your review comments. Please check.

File: library/src/lib/menu/menu.component.ts (hunk 1 above)
Comment (kavya-b) [reply library/src/lib/menu/menu.component.ts]: done

File: library/src/lib/menu/menu.component.ts (hunk 2 above)
Comment (kavya-b) [reply library/src/lib/menu/menu.component.ts]: done

**Pull request merged**